onetrain.py

```diff
@@ -244,18 +288,21 @@ def set_config(args: TrainArgs):
 
 def train(args: TrainArgs):
     if not args.train:
+        log.info("returning from train because no train arg")
         return
 
-    set_path(args)
+    log.info("setting path") # set path causes logging issues
+    #set_path(args)
+    log.info("setting buckets")
     buckets(args)
 
-    from modules.util.callbacks.TrainCallbacks import TrainCallbacks # pylint: disable=import-error
-    from modules.util.commands.TrainCommands import TrainCommands # pylint: disable=import-error
-    from modules.trainer.GenericTrainer import GenericTrainer # pylint: disable=import-error
+    info.concept = args.concept
+
+    #log.info("imported training modules")
 
     def train_progress_callback(p, max_sample, max_epoch):
         ts = time.time()
-        info.progress = p
+        info.progress = p.filename_string()
```

Comment:
this obfuscates a lot of information.
if you want a filename, add filename to info object, don't replace main object.

Reply:
I saw it is in this format: f"{self.global_step}-{self.epoch}-{self.epoch_step}"
so it returns all the information about the progress and is the only read function of the class

Reply:
For reference: https://github.com/Nerogar/OneTrainer/blob/master/modules/util/TrainProgress.py#L24